### ISSUE TITLE
Cannot replicate the cumulative frequency response of the SincNet filters

### ISSUE THREAD

**Issue by tuananh0305** [mravanelli/SincNet#88]
Hi Marco,

Thanks for sharing your great work!

I am trying to replicate the cumulative response of the SincNet filter like the figure 3 in the paper (https://arxiv.org/pdf/1808.00158.pdf).

I load your pre-trained model and computed the Fourier transforms of all the learned filters like figure below.
![ask_Figure_2](https://user-images.githubusercontent.com/35039289/81115154-8e2a8500-8f23-11ea-95f7-fcafd0132250.png)

Then I averaged the sum of all frequency response of the learned filters to get the cumulative response of the SincNet filter like figure below
![ask_Figure_1](https://user-images.githubusercontent.com/35039289/81115740-7bfd1680-8f24-11ea-9609-a14d72807d78.png)

What is your opinion about this problem?

**Comment by mravanelli** [comment]
Hi,
I'm aware of it. In practice, after the submission we did a code
refactoring and for some reason that I have to investigate better (probably
due to filter amplitude normalization) the gradient is smaller than before
and the filters are more static than those of the original version. I will
work on that to find exactly the modification that caused that. For now, I
think you can add a separate optimizer for the sincnet layer and use higher
learning rate for it. This will encourage the filters to change more and
eventually reach some configuration like the one shown in the
original paper.

**Comment by tuananh0305** [comment]
Hi Marco,

Thanks for your quick response!

I will try your suggestion. Thank you!

**Comment by JLC7** [comment]
I am getting small changes to the input layer. Any luck on using larger learning rates?
